### PR TITLE
refactor: consistently use one uuid library

### PR DESCRIPTION
**Pull request by kruskall**
## Proposed commit message

consistently use one uuid library

the codebase is using a mix of github.com/google/uuid and github.com/gofrs/uuid Update the codebase to use github.com/gofrs/uuid consistently and bump to the current major version.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

- Depends on https://github.com/elastic/elastic-agent-system-metrics/pull/163
- Depends on https://github.com/elastic/elastic-agent-client/pull/118

### CODE REVIEW

**Comment by efd6** [comment]
Is there a good argument for one over the other? The APIs look pretty similar, but there are some aspects of each that are not in the other, so one is not a drop in replacement for the other.

**Comment by efd6** [comment]
> Can you expand on this ?

- The constructor sets differing is the most obvious thing; there is no `gofrs/uuid.New`.
- The errors that are returned are different (I think gofrs is stronger here, but they differ, so use of the error values needs to be checked).
- There are a few method set differences, so interface satisfaction will differ which may not be noticeable for dynamically asserted interface behaviour.

**Review by belimawr — COMMENTED** [review]
So far the PR looks great! I don't have any strong preference about which library to use, but I like the fact that `github.com/gofrs/uuid` is community maintained.

**Comment by leehinman** [comment]
@kruskall do you think it would be worth it to add the google/uuid to the gomodguard section of `.golangci.yml` file?  That way CI can help us stay with one library.

**Comment by aleksmaus** [comment]
> I don't have any strong preference about which library to use, but I like the fact that `github.com/gofrs/uuid` is community maintained.

github.com/gofrs/uuid is a way to go

**Review by nfritts — APPROVED** [review]

File: libbeat/monitoring/inputmon/input.go [hunk 1]
```diff
@@ -59,7 +59,7 @@ func NewInputRegistry(inputType, id string, optionalParent *monitoring.Registry)
 	// logs during support interactions.
 	log := logp.NewLogger("metric_registry")
 	// Make an orthogonal ID to allow tracking register/deregister pairs.
-	uuid := uuid.New().String()
+	uuid := uuid.Must(uuid.NewV4()).String()
```
Comment (efd6) [inline libbeat/monitoring/inputmon/input.go:62]: AFAICS `crypto/rand.Reader` hides EAGAIN so this should be OK.

**Review by efd6 — APPROVED** [review]
It's a pity that this is mixed up with lint clean up.

File: libbeat/cmd/instance/beat.go [hunk 2]
```diff
@@ -280,7 +275,7 @@ func NewBeat(name, indexPrefix, v string, elasticLicensed bool, initFuncs []func
 			ID:              id,
 			FirstStart:      time.Now(),
 			StartTime:       time.Now(),
-			EphemeralID:     eid,
+			EphemeralID:     metricreport.EphemeralID(),
```
Comment (pkoutsovasilis) [inline libbeat/cmd/instance/beat.go:278]: on the left `metricreport.EphemeralID()` is essentially checked if it is a valid uuid with `uuid.FromString`, however, this is removed now. Any reason that this check became redundant?

**Review by pkoutsovasilis — APPROVED** [review]
A single nit about a check that might have become redundant, but this LGTM

File: libbeat/cmd/instance/beat.go (hunk 2 above)
Comment (pkoutsovasilis) [reply libbeat/cmd/instance/beat.go:278]: ok cool to know I was just double-checking that this was intentional and not something missed 🙂

**Review by andrewkroh — APPROVED** [review]

**Review by leehinman — APPROVED** [review]

**Review by axw — APPROVED** [review]

**Review by kaiyan-sheng — APPROVED** [review]

**Pull request merged**